### ISSUE TITLE
ProActive applications don't terminate without using a System.exit when using RMI

### ISSUE THREAD

**Issue by activeeon-bot** [ow2-proactive/programming#263]
<a href="https://jira.activeeon.com/browse/PROACTIVE-372" title="PROACTIVE-372">Original issue</a> created by <a href="mailto:fabien.viale_AT_activeeon.com">Fabien Viale</a> on 18, Jun 2008 at 17:26 PM - PROACTIVE-372

<hr />

self explanating

The reason behind is related to what is explained here:
http://forum.java.sun.com/thread.jspa?threadID=169975&messageID=519735

Exported RMI remote object (Active Objects, GCM deployment, Runtime, Half bodies) are not unexported. Thus, the RMI Reaper thread terminates only after the very long default RMI timeout (10 minutes).

Active Objects' remote objects should be unexported at the end of their life cycle (terminate method, after automatic continuations have been resolved).

Runtime and GCM deployment Remote Objects should be unexported when calling the PALifeCycle.exitSuccess() or a method like  PALifeCycle.terminate()

Calling a System.exit(0) is sufficient in many cases, but their are cases (for example if ProActive runs inside an Web or application server) where it would be needed to clear ProActive references without killing the whole environment.

**Comment by activeeon-bot** [comment]
<a href="https://jira.activeeon.com/browse/PROACTIVE-372?focusedCommentId=18514">Original comment</a> posted by <a href="mailto:youri.bonnaffe_AT_activeeon.com">Youri Bonnaffe</a> on 12, Dec 2014 at 12:54 PM

<hr />

Marking old issues as won't fix, either because we don't support it anymore, or it is needed or it has been there for too long.

Feel free to reopen if needed.
